src/asm/fstack.c

```diff
@@ -511,9 +513,9 @@ void fstk_RunFor(char const *symName, int32_t start, int32_t stop, int32_t step,
 	uint32_t count = 0;
 
 	if (step > 0 && start < stop)
-		count = (stop - start - 1) / step + 1;
+		count =  (uint32_t)((uint64_t)((int64_t)stop - start - 1) / step) + 1;
 	else if (step < 0 && stop < start)
-		count = (start - stop - 1) / -step + 1;
+		count = (uint32_t)((uint64_t)((int64_t)start - stop - 1) / -step) + 1;
```

Comment:
If you're going to be careful about UB, remember that `-step` is UB if `step` is `INT32_MIN` (assuming `int32_t` is at least as wide as `int`, which in practice it will almost always be). You should cast `step` to `int64_t` _before_ negating it.